### PR TITLE
Fix anchors links and github page refresh

### PR DESCRIPTION
**Pull request by KamiKillertO**
This is a fix for #30.
I've removed the settings `locationType` `hash` for productions build. Now links are generated without a `#` inside and make it possible to use to anchors links in pages.

Changing the settings `locationType` breaks the page refresh on github, to fix this problem I've used the same solution as [ember-cli-addon-docs](https://github.com/ember-learn/ember-cli-addon-docs/). I've created a [`404.html` page](https://github.com/ember-a11y/a11y-demo-app/pull/31/files#diff-e213ad74dd5c0e4406e4277d5760ae53) this page is displayed by GitHub when you try to access a page that doesn't exist. This page contains the script provided by this project [spa-github-pages](https://github.com/rafrex/spa-github-pages) which takes the currentUrl, move the part causing the 404 into a query param and redirect to the github index page. In the [index.html]() page there's another script that will take the URL and move back the parts that are in the query params into the URL location.

Basically this what happen :

1. Someone try access https://ember-a11y.github.io/a11y-demo-app/foo
2. The page does not exist so github render the 404.html page
3. The script in the page transform the URL into https://ember-a11y.github.io/a11y-demo-app?q=/foo
4. User is redirected to https://ember-a11y.github.io/a11y-demo-app
5. The script in the index.html will transform the URL from https://ember-a11y.github.io/a11y-demo-app?q=/foo into https://ember-a11y.github.io/a11y-demo-app/foo using `window.history.replaceState`
6. Because the script in index.html use `window.history.replaceState` ember router will redirect to correct page

### CODE REVIEW

**Comment by KamiKillertO** [comment]
Changing the `locationType` setting only will break the page refresh. I've the look to ember-cli-addons-docs and they solve this problem by adding a 404 page with the a redirection script to the index.html page

**Comment by rajasegar** [comment]
Thanks for working on this @KamiKillertO 
I think it got messed up during migration, will it work if we have set `locationType:auto`

**Review by rajasegar — APPROVED** [review]
